packages/salesforcedx-vscode-core/src/services/sourceStatusSummary.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Row,
+  Table
+} from '@salesforce/salesforcedx-utils-vscode/out/src/output';
+import { StatusOutputRow } from '@salesforce/source-tracking';
+
+type StatusActualState = 'Deleted' | 'Add' | 'Changed' | 'Unchanged';
+type StatusOrigin = 'Local' | 'Remote';
+
+interface StatusResult {
+  state: string;
+  fullName: string;
+  type: string;
+  filePath?: string;
+  ignored?: boolean;
+  conflict?: boolean;
+  actualState?: StatusActualState;
+  origin: StatusOrigin;
+}
+
+interface FormattedStatusResult {
+  state: string;
+  fullName: string;
+  type: string;
+  filePath?: string;
+  ignored?: string;
+}
+
+export class SourceStatusSummary {
```

Comment:
The functionality to convert the raw response from STL into the formatted table-like output was mostly lifted from plugin-source.  I put it into two objects: SourceStatusSummary, which takes the raw output and produces a formatted string, and StatusResultsTable, which is used internally by SourceStatusSummary to build a table from the raw response.